### PR TITLE
feat(token): add new $text-error

### PR DESCRIPTION
**Pull request by aagonzales**
Related: https://github.com/carbon-design-system/carbon/issues/5024

- Added new `$text-error` token
- Updated value of `$support-01` in the G90 theme.

### CODE REVIEW

File: packages/themes/src/g100.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ export const text02 = gray30;
 export const text03 = gray60;
 export const text04 = white;
 export const text05 = gray50;
+export const textError = red40;
```
Comment (joshblack) [inline packages/themes/src/g100.js:61]: I think we'll need to add `red40` in the import block at the beginning of file, maybe adding it before `red50`?

```diff
import {
  // Blue
  blue30,
  blue40,
  blue50,
  blue60,
  blue80,

  // Gray
  gray10,
  gray30,
  gray50,
  gray60,
  gray70,
  gray80,
  gray90,
  gray100,

  // Support
  green40,
  green50,
  yellow,
+ red40,
  red50,
  red60,
  red80,
  purple40,

  // Constants
  white,

  // Tools
  rgba,
} from '@carbon/colors';
```

File: packages/themes/src/g90.js [hunk 2]
```diff
@@ -58,6 +58,7 @@ export const text02 = gray30;
 export const text03 = gray60;
 export const text04 = white;
 export const text05 = gray50;
+export const textError = red30;
```
Comment (joshblack) [inline packages/themes/src/g90.js:62]: I think `red30` and `red40` will need to be added above, similar to that note above!

**Review by joshblack — CHANGES_REQUESTED** [review]
Seems great! Left some comments for some import issues we might run into

File: packages/themes/src/g90.js (hunk 2 above)
Comment (aagonzales) [reply packages/themes/src/g90.js:62]: Ok, I knew there was probably something else I needed to do.

File: packages/themes/src/g90.js [hunk 3]
```diff
@@ -28,6 +28,8 @@ import {
   green40,
   green50,
   yellow,
+  red30,
+  red40,
   red50,
```
Comment (tw15egan) [inline packages/themes/src/g90.js]: Since we no longer use `red50` in this file, we can remove the import

**Comment by tw15egan** [comment]
Does the new variable also have to be added towards the bottom of `tokens.js`?    
    https://github.com/carbon-design-system/carbon/blob/4cae6f2e1248a1b5fa650ab63da914e44c573549/packages/themes/src/tokens.js#L177

Not quite sure what the `unstable__meta` is used for

**Comment by joshblack** [comment]
@tw15egan good catch, `unstable__meta` is for the Sketch plugin so that the colors are placed in the correct grouping 👍

**Review by tw15egan — APPROVED** [review]
Thanks for adding these in, Anna! 👍 💪

**Comment by asudoh** [comment]
@joshblack Just in case you have any further comments - Thanks!

**Review by joshblack — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @aagonzales!

**Comment by RoystonS** [comment]
Will you also be updating the `metadata.yml` file to reflect these changes?

**Pull request merged**